### DISCUSSION TITLE
CUDA OpenCV on Fedora

### DISCUSSION THREAD

**Discussion by neurovish** [Breakthrough/DVR-Scan#175] [Q&A]
I finally got OpenCV compiled and working on Fedora. It seems like every guide/discussion on the topic is about Ubuntu...in the end, I think it would've been far easier to just rebuild my system with Ubuntu. It generally works, but I've hit a bit of a wall. Using the stock dvr-scan[opencv] from pip, the output motion AVIs are encoded with xvid ("Xvid MPEG-4 Video (XVID)"), but using the CUDA OpenCV I compiled, the output motion AVIs are encoded with divx and results in a much lower quality video.

My guess is that it is a gstreamer vs ffmpeg thing and I'm missing some library or plugin, but I've tried all the ones I found in the repository that make sense and no luck. Anybody know why it might be behaving this way?

**Comment by Breakthrough** [comment]
That's pretty odd, does setting the `opencv-codec` option in a config file do anything?  This might have to do with some other OpenCV feature flags enabled at compile time.

That being said, consider using output mode `-m ffmpeg` since that gives you complete control over bitrate and other encoding parameters.

**Reply by neurovish** [reply]
Ahhh, I might go back and try that, but I ended up forging ahead and compiled my own ffmpeg toolchain after blowing up several different attempts. I never could manage to get opencv to compile with CUDA and ffmpeg using the Fedora packages. Once I got ffmpeg back in the mix, everything is sorted out.

Just so it is noted somewhere before I forget, I followed https://trac.ffmpeg.org/wiki/CompilationGuide/Centos to install ffmpeg, but built everything after yasm and nasm as shared instead of static. With an entry in ld.so.conf pointing to the new ffmpeg libraries and setting PKG_CONFIG_PATH to point to the directory with the new .pc files, the opencv build recognized everything.
